dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/helpers/calculate-initial-utilisation-and-fixed-fee.ts

```diff
@@ -0,0 +1,69 @@
+import { calculateInitialUtilisation, isValidDate } from '@ukef/dtfs2-common';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { calculateFixedFee } from './calculate-fixed-fee';
+
+export type RequiredParams = {
+  value?: number | null;
+  interestPercentage?: number | null;
+  dayCountBasis?: number | null;
+  coverStartDate?: string | Date | null;
+  coverEndDate?: string | Date | null;
+};
+
+/**
+ * checks that all required values are present
+ * checks that cover start and end dates are in date format
+ * @param value
+ * @param interestPercentage
+ * @param dayCountBasis
+ * @param coverStartDate
+ * @param coverEndDate
+ * @returns true if all values are present and in the correct format
+ */
+export const hasRequiredValues = ({ value, interestPercentage, dayCountBasis, coverStartDate, coverEndDate }: RequiredParams): boolean =>
+  Boolean(
+    value &&
+      coverStartDate &&
+      coverEndDate &&
+      interestPercentage &&
+      dayCountBasis &&
+      isValidDate(new Date(coverStartDate)) &&
+      isValidDate(new Date(coverEndDate)),
+  );
+
+/**
+ * calculateInitialUtilisationAndFixedFee
+ * calculates initial utilisation and fixed fee for a facility
+ * gets the facility from the tfm facilities repo
+ * calculates initial utilisation and fixed fee from facility values
+ * returns the calculated values
+ * @param {String} facilityId
+ * @returns {Object} fixedFee and utilisation values
+ */
+export const calculateInitialUtilisationAndFixedFee = async (facilityId: string) => {
+  const tfmFacility = await TfmFacilitiesRepo.findOneByUkefFacilityId(facilityId);
+
+  if (!tfmFacility?.facilitySnapshot) {
+    throw new Error(`TFM facility ${facilityId} could not be found`);
+  }
+
+  const { value, coverStartDate, coverEndDate, interestPercentage, dayCountBasis } = tfmFacility.facilitySnapshot;
```

Comment:
does this require a seperate ticket?

Reply:
I *think* we don't need to because the update that will be generated in the keying sheet report will be accurate for what we should update the value to. While we are generating this late all we want this to be is the 'initial' values that the facility would have at it's inception when it gets into ACBS for the first time

Reply:
Ahh I see, that makes sense. I guess on the ticket about facility value changing and manual updates we have comments already about finding out if PDC would manually update fixed fee in that case and this is actually similar to that.